src/inttest/java/com/faforever/api/utils/OAuthHelper.java

```diff
@@ -1,66 +1,67 @@
 
+    Set<FafRole> roles = user.getUserGroups().stream()
+      .flatMap(userGroup -> userGroup.getPermissions().stream())
+      .map(permission -> new FafRole(permission.getTechnicalName()))
+      .collect(Collectors.toSet());
 
-  public RequestPostProcessor addBearerToken(Set<String> scope, Set<String> authorities) {
-    return addBearerToken(null, null, scope, authorities);
-  }
+    var fafScopes = scopes.stream().map(FafScope::new).toList();
 
-  public RequestPostProcessor addBearerToken(Integer userId, String userName, Set<String> scope, Set<String> authorities) {
-    return mockRequest -> {
-      Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-      OAuth2Request oauth2Request = createOAuth2Request(scope, authorities);
-      OAuth2Authentication oauth2auth = new OAuth2Authentication(oauth2Request, authentication);
-      DefaultOAuth2AccessToken token = (DefaultOAuth2AccessToken) tokenServices.createAccessToken(oauth2auth);
-      token.setAdditionalInformation(new HashMap<>(token.getAdditionalInformation()));
-      if (userId != null) {
-        token.getAdditionalInformation().put(USER_ID_KEY, userId);
-      }
+    return authentication(new FafAuthenticationToken(userId, fafScopes, roles));
+  }
 
-      if (userName != null) {
-        token.getAdditionalInformation().put(USERNAME, userName);
-      }
+  public RequestPostProcessor addBearerToken(@NotNull Set<String> scope, @NotNull Set<String> authorities) {
+    return addBearerToken(null, scope, authorities);
+  }
 
-      token = (DefaultOAuth2AccessToken) jwtAccessTokenConverter.enhance(token, oauth2auth);
+  public RequestPostProcessor addBearerToken(
+    Integer userId,
+    @NotNull Set<String> scopes,
+    @NotNull Set<String> roles
+  ) {
+    var fafScopes = scopes.stream().map(FafScope::new).toList();
+    var fafRoles = roles.stream().map(FafRole::new).toList();
 
-      // Set Authorization header to use Bearer
-      mockRequest.addHeader("Authorization", "Bearer " + token.getValue());
-      return mockRequest;
-    };
+    return authentication(new FafAuthenticationToken(userId, fafScopes, fafRoles));
   }
 
-  @NotNull
-  private OAuth2Request createOAuth2Request(Set<String> scope, Set<String> authorities) {
-    List<GrantedAuthority> grantedAuthorities = authorities == null ? null : AuthorityUtils.createAuthorityList(authorities.toArray(new String[0]));
-    return new OAuth2Request(null, "00000000-0000-0000-0000-000000000000", grantedAuthorities, true, scope, null, null, null, null);
+  /*
+
+  public RequestPostProcessor addBearerToken(
+    Integer userId,
+    String userName,
+    @NotNull Set<String> scopes,
+    @NotNull Set<String> authorities
+  ) {
+    var mappedScopes = scopes.stream().map(FafScope::new).toList();
+    var mappedRoles = scopes.stream().map(FafRole::new).toList();
+
+    return authentication(new FafAuthenticationToken(userId, mappedScopes, mappedRoles));
   }
+   */
```

Comment:
Why leave commented out code?

Reply:
Because I played around a billion times forth and back and obviously didn't clean up afterwards :stuck_out_tongue: